### PR TITLE
HPol Pull Request

### PR DESCRIPTION
**Pull request by unmovingcastle**
removed vector original_location as well as int t ; will run a test to see if this affects anything. 
This is a dummy commit to help Jason learn how to add reviewers

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,9 +27,13 @@
...
+*.x
 *.exe
 *.out
 *.app
 
 # Folders and Repositories
 *.vs
+
+# Makefiles
+*Makefile
```
Comment (ezio-melotti) [inline .gitignore:39]:
```suggestion
Makefile
```
Unless the `Makefile` has some prefix in the filename, the `*` can be removed.

File: GA/HeaderFiles/ConstraintHPol.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  else
+  {
+    intersect = false;
+  }
+
+  // Return interesect condition
+  return intersect;
+}
+
+#endif
```
Comment (ezio-melotti) [inline GA/HeaderFiles/ConstraintHPol.h:29]: There should be an empty line at the end of the file.  You should be able to configure your editor to add it automatically when you save the file (if it's not there already).  Checking the option to remove trailing whitespace on save is also a good idea.

We also discussed coding conventions regarding indentation and `{}` placements, but we still don't have a system in place to check/fix it automatically.

File: GA/HeaderFiles/DataRead.h [hunk 3]
```diff
@@ -20,50 +20,45 @@ void DataRead(vector<vector<vector<float> > >& dna_input,
...
+  int csv_file_size = population * sections;
+  ifstream generationDNA("generationDNA.csv");
   string csv_content[csv_file_size];
-  string str_to_dbl;
+  string str_to_dbl;                            // string to double
 
-  // This loop reads through the .csv file line by line
-  // If we're in data (past line 9), it reads in the line
+  string dumpster;                              // getting rid of headers
+  for (int i=0; i<dna_garbage_end; i++) getline(generationDNA, dumpster);
```
Comment (ezio-melotti) [inline GA/HeaderFiles/DataRead.h:30]:
```suggestion
  for (int i=0; i<dna_garbage_end; i++) {
      getline(generationDNA, dumpster);
  }
```
It's generally better to keep one expression per line, and to always include `{}` even when the body of the loop has a single expression.

File: GA/HeaderFiles/DataRead.h [hunk 4]
```diff
@@ -20,50 +20,45 @@ void DataRead(vector<vector<vector<float> > >& dna_input,
...
 
-  ifstream fitnessScores;
-  fitnessScores.open("fitnessScores.csv");
-  string fitness_read[population + 2];
+  ifstream fitnessScores("fitnessScores.csv");
+  string fitness_read[population];
 
-  for (int i = 0; i < (population + 2); i++)
+  int num_header = 2;                           // getting rid of headers
+  for (int i=0; i<num_header; i++) getline(fitnessScores,dumpster);
```
Comment (ezio-melotti) [inline GA/HeaderFiles/DataRead.h:53]:
```suggestion
  for (int i=0; i<num_header; i++) {
      getline(fitnessScores,dumpster);
  }
```

File: GA/HeaderFiles/DataWrite.h [hunk 5]
```diff
@@ -22,9 +22,11 @@ void DataWrite(std::vector<std::vector<std::vector<float> > >& dna_output,
   // Write the header
   ofstream generationDNA;
   generationDNA.open("generationDNA.csv");
-  generationDNA << "Hybrid of Roulette and Tournament -- Thanks to Cal Poly / Jordan Potter" << "\n";
+  generationDNA << "Hybrid of Roulette and Tournament -- "
+                   "Thanks to Cal Poly / Jordan Potter" << "\n";
   generationDNA << "Author was David Liu" << "\n";
-  generationDNA << "Notable contributors: Julie Rolla, Hannah Hasan, and Adam Blenk" << "\n";
+  generationDNA << "Notable contributors: Julie Rolla, Hannah Hasan, "
+                   "and Adam Blenk" << "\n";
```
Comment (ezio-melotti) [inline GA/HeaderFiles/DataWrite.h:29]: PR should generally focus on a single feature/issue and shouldn't include extraneous changes (unless it's a minor change and you are already touching that part of the code).  If, for example, this PR gets rejected or reverted, this unrelated change will also get lost.

**Review by ezio-melotti — COMMENTED** [review]
I left a few comments.  In general:
* PRs should focus on implementing a single feature or fixing a single bug (regardless of the number of commits it takes to do it), which might described on a separate Issue (but not necessarily).
* Unrelated changes (e.g. reformatting) should be avoided (unless they are minor changes on code you are already touching).
* Force-pushing should be avoided, since it makes it harder to review and if someone already pulled your branch it might create issues.  It's fine to close/reject a PR and open a new one.
* You should be able to configure your editor to remove trailing whitespace and add a newline at the end of each file automatically.

I noticed that some of the file are reformatted to shorten long lines.  Is this something you did manually or did you use some tool to do it?

File: GA/HeaderFiles/DataWrite.h (hunk 5 above)
Comment (unmovingcastle) [reply GA/HeaderFiles/DataWrite.h:29]: Hi Ezio,
Sorry I didn't know it's going to send notification to you each time I make an edit.
I have reverted the main branch to what its original state (before any HPol commits) and then squashed the HPol branch into one commit.
I'll make one last edit on HPol following the comments above and then stop editing HPol until we are ready to merge HPol back to main. 
In the meantime I'll just make a separate branch to poke at.

Comment (ezio-melotti) [reply GA/HeaderFiles/DataWrite.h:29]: Don't worry about the notifications, I can tweak the settings to be notified about what I want (and I get plenty of notifications every day anyway :).

The usual workflow is:
1. create a local branch with some commits
2. push the branch on GitHub (`git push origin your-branch`) and create a PR (it gives you a link to create it after the push)
3. wait for reviews and comments
4. add more commits to the local branch to address the reviews/comments
5. push them to GitHub (`git push origin your-branch`, without forcing/squashing/rebasing) -- the PR will update automatically to include the new commit
6. repeat steps 3-5 until the PR is approved
7. click on the squash merge button on GitHub to merge the PR on `main`

Comment (unmovingcastle) [reply GA/HeaderFiles/DataWrite.h:29]: Understood. Thank you!
